### DISCUSSION TITLE
Dispatching a sublattice

### DISCUSSION THREAD

**Discussion by scottwn** [AgnostiqHQ/covalent#788] [Q&A] [answered]
Consider this code
```
@ct.electron
def new_func(a, b, c, d, e):
    time.sleep(10)
    return a + b + c + d + e


@ct.electron
@ct.lattice
def work_func(a, b, c):
    return new_func(a, b, c, d=4, e=5)

dispatch_id = ct.dispatch(work_func)(5, 6, 7)
print(dispatch_id)
```

Running this will produce an error like this
```
Traceback (most recent call last):
  File "example_db.py", line 246, in <module>
    dispatch_id = ct.dispatch(lattice)(*args)
  File "/Users/scott/covalent/covalent/_dispatcher_plugins/local.py", line 83, in wrapper
    lattice.build_graph(*args, **kwargs)
AttributeError: 'function' object has no attribute 'build_graph'
```

I expect this is because I'm trying to dispatch a sublattice. If that's the case, we should have a check in the code that raises an informative error so users don't see the AttributeError.

Loosely related to #760

**Comment by FyzHsn** [comment, marked as answer]
That's a really good point, we should probably return an error saying, "Only lattices can be dispatched"

**Reply by santoshkumarradha** [reply]
Good point, I think the error is obvious that this is an electron type and not a lattice type. The error should be that "only lattice can be dispatched"

**Reply by santoshkumarradha** [reply]
And if you want to dispatch this, we need to top it off with a lattice decorator. ( the top most decorator defines the final type)
